src/http.cc

```diff
@@ -1949,6 +2092,48 @@ HttpStateData::httpBuildRequestHeader(HttpRequest * request,
     strConnection.clean();
 }
 
+void
+HttpStateData::makeUpgradeHeaders(HttpHeader &hdr_out)
+{
+    const HttpHeader &hdr_in = request->header;
```

Comment:
We do not need the virgin request any more but the adapted request. 
The a30030c commit allow ICAP server to modify Upgrade header, Squid should use the adapted request when computes outgoing Upgade header.

Reply:
The eCAP before a30030c commit received hop-by-hop header. But even if eCAP modified the Upgrade header we taken in account only the Upgrade header from virgin request.

We need to allow adaptation services (eCAP or ICAP) to check and modify Upgrade header so the commit  a30030c  does the followings:
  - Include Upgrade header to the headers sent to the ICAP (eCAP already receive it)
  - Use the adapted HttpRequest instead of the virgin to build the final outgoing Upgrade header.